syntax = "proto3";

package kvstore;

service KeyValueStore {
  rpc Set (SetRequest) returns (SetResponse) {}
  rpc Get (GetRequest) returns (GetResponse) {}
  rpc GetPrefix (GetPrefixRequest) returns (GetPrefixResponse) {}
}

message SetRequest {
  string key = 1;
  bytes value = 2;
}

message SetResponse {
  bool status = 1;
  string message = 2;
}

message GetRequest {
  string key = 1;
}

message GetResponse {
  bool found = 1;
  bytes value = 2;
}

message GetPrefixRequest {
  string prefix = 1;
}

message GetPrefixResponse {
  repeated bytes values = 1;
}